smaug/core/network_builder.cpp

```diff
@@ -108,15 +117,8 @@ static void createAndAddOperator(const NodeProto& node,
     dout(0) << "Adding " << name << " (" << OpType_Name(type) << ").\n";
 
     if (type == OpType::Data) {
-        // Find the tensor data from the tensor data array.
-        TensorData tensorData;
-        for (int i = 0; i < tensorDataArray.data_array_size(); i++) {
-            if (tensorDataArray.data_array(i).name() ==
-                node.input_tensors(0).name()) {
-                tensorData = tensorDataArray.data_array(i);
-                break;
-            }
-        }
+        const TensorData& tensorData = searchTensorDataArray(
```

Comment:
If you're getting rid of data ops, shouldn't this section be entirely deleted?

Reply:
Some of our older models were using this data op, so I think it's worth keeping it.